Sources/core/objects/Human.cpp

```diff
@@ -125,7 +125,16 @@ void Human::processGUImsg(const Message2 &msg)
     if (msg.type == MessageType::MESSAGE)
     {
         std::string text = obj["text"].toString().toStdString();
-        GetGame().GetChat().PostWords(name, text, owner.Id());
+        if (text.find("me") == 0 || text.find("*") == 0)
+        {
+            size_t length = text.find("me") == 0 ? std::string("me").length() : std::string("*").length();
```

Comment:
Also, does your code work properly with spaces
`me smiles`, `mesmiles`, `*smiles` and `* smiles` should be 'Morgan James smiles' with one space between name and text. So you may want to actually make more prefixes like "me", "me ", "*" and "* " and replacing them with name plus one space.